ci/test/02_run_container.sh

```diff
@@ -30,6 +30,10 @@ if [ -z "$DANGER_RUN_CI_ON_HOST" ]; then
   docker volume create "${CONTAINER_NAME}_previous_releases" || true
 
   if [ -n "${RESTART_CI_DOCKER_BEFORE_RUN}" ] ; then
+    # Kill harder, prevents "The container name "/ci_..." is already in use by container"
+    docker stop "$(docker ps -a -q)" || true
```

Comment:
What is "kill harder"? Containers are already being `--all --force` removed below:
```bash
-a, --all                   Remove all containers
-f, --force                 Force removal of a running or unusable container
```
so they shouldn't need to be "stopped" beforehand?

Also, shouldn't this be `podman`?

Reply:
It was intentionally vague, because I might just be doing something else wrong. I'll read @maflcko's suggestion.

Reply:
Oh, this is only available in Noble? In that case I might just improve the code comment (including `sleep`) and add a note that can go away with Noble.

Reply:
I installed Podman 4.8.3 from source on my machine. However I don't understand what you mean by `run --replace`. It seems we're only using podman in the `RESTART_CI_DOCKER_BEFORE_RUN`, but still call `docker run` everywhere. Can we just swap `docker run` with `podman run` or is it more complicated?

Reply:
I don't have docker installed on any machine I use for CI. Only podman, and a docker -> podman alias (podman-docker).

Reply:
Mmm, I tried something similar in de26cbc79230f64d7747b8b88f4dee71ebd37e1b by adding a `DOCKER_RUN` variable and setting it to `podman run --replace`. But this fails with `Error: short-name resolution enforced but cannot prompt without a TTY`. Are you passing any other arguments in your alias?

Reply:
> Are you passing any other arguments in your alias?

The alias came from:
```bash
apt install podman-docker
# podman-docker is already the newest version (4.8.3+ds1-2).
docker --version
# podman version 4.8.3
podman --version
# podman version 4.8.3
```

Reply:
podman-docker is not documented in the manual installation... https://podman.io/docs/installation#building-from-source

Let's see how I can install this without totally messing up my OS :-)

Reply:
That base64 decodes to:

```sh
#!/usr/bin/env bash

echoerr() { echo "$@" 1>&2; }

echoerr " (:::docker->)podman ${@}"

podman "${@}"
```

That doesn't seem any different from my `DOCKER_RUN="podman run --replace"` alias.

Reply:
Or I guess it's the `echoerr` doing some magic

Reply:
Yes, Docker is installed in rootless mode: https://docs.docker.com/engine/security/rootless/

Forgot to link to the error: https://cirrus-ci.com/task/5247017476161536?logs=ci#L261

The "short-name resolution" error makes no sense to me based on what I can find about it, since the container URL includes `docker.io` (`docker.io/amd64/ubuntu:22.04` in the above example).

> you can't install podman-docker without removing docker

Ok, let's not go that route, since Docker currently works.

> because the cleanup is done with the podman command, which can not clear docker containers

Ah, so that explains why I needed 7cdb75d572e825b4c0cf74a516f912f94807853f.

> This should fix the issue you are trying to "fix" in this commit.

Why do we add a dependency on podman at all? Given that `docker stop "$(docker ps -a -q)" || true` works. I guess it's less clean because it cleans all docker processes run by the user. But is there some other benefit? Is the plan to move to podman entirely?

Reply:
Oh, I see where my confusion comes from. I thought podman was just a wrapper for Docker. But it's actually an alternative which can run the same containers. Why not switch over entirely to Podman? It's quite confusing that we're using both.

Reply:
I nuked Docker in favor of Podman. So far it seems to work. I dropped 7cdb75d572e825b4c0cf74a516f912f94807853f.

Reply:
Not specifically related to the issue at hand: the new podman desktop app is pretty smart, it can show you containers running via docker and podman (and kubernetes clusters) all in the same UI: https://podman-desktop.io/